### PR TITLE
Fix Incorrect NSFW Status on Channels

### LINKED ISSUE #2498: 🐛 Bug: message.channel.nsfw
if the channel is named `nsfw`, the function `message.channel.nsfw` returns `false`.

Versions:

Discord.js - 11.3.2

Node.js - 8.9.4

npm - 5.6.0

Images:

coolimage1
coolimage2

### PR DESCRIPTION
**Pull request by lloti**
**Please describe the changes this PR makes and why it should be merged:**
To fix #2498. According to the guys over at the Discord API server, this is the proper way to check a channel's NSFW state.

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/TextChannel.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ class TextChannel extends GuildChannel {
      * @type {boolean}
      * @readonly
      */
-    this.nsfw = Boolean(data.nsfw);
+    this.nsfw = data.nsfw || this.name === 'nsfw' || this.name.startsWith('nsfw-');
```
Comment (dragonfire535) [inline src/structures/TextChannel.js]: or you could ` /^nsfw(-|$)/.test(this.name);` 🤔

**Comment by iCrawl** [comment]
See #2498

**Comment by appellation** [comment]
This should be implemented directly in the `_patch` method.  It seems pointless to make this a getter since the property will only change after a patch.

File: src/structures/TextChannel.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ class TextChannel extends GuildChannel {
      * @type {boolean}
      * @readonly
      */
-    this.nsfw = Boolean(data.nsfw);
+    this.nsfw = data.nsfw || /^nsfw(-|$)/.test(this.name);
```
Comment (MrJacz) [inline src/structures/TextChannel.js]: The regex should be a variable instead of creating it every time the function is called

**Comment by Lewdcario** [comment]
@lloti What's the status on the PR? Essentially waiting on @appellation's suggestion to be implemented.

**Pull request merged**